skimage/segmentation/morphsnakes.py

```diff
@@ -0,0 +1,442 @@
+    lambda2 : float, optional
+        Weight parameter for the inner region. If `lambda2` is larger than
+        `lambda1`, the inner region will contain a larger range of values than
+        the outer region.
+    iter_callback : function, optional
+        If given, this function is called once per iteration with the current
+        level set as the only argument. This is useful for debugging or for
+        plotting intermediate results during the evolution.
+
+    Returns
+    -------
+    out : (M, N) or (L, M, N) array
+        Final segmentation (i.e., the final level set)
+
+    See also
+    --------
+    circle_level_set, checkerboard_level_set
+
+    Notes
+    -----
+
+    This is a version of the Chan-Vese algorithm that uses morphological
+    operators instead of solving a partial differential equation (PDE) for the
+    evolution of the contour. The set of morphological operators used in this
+    algorithm are proved to be infinitesimally equivalent to the Chan-Vese PDE
+    (see [1]_). However, morphological operators are do not suffer from the
+    numerical stability issues typically found in PDEs (it is not necessary to
+    find the right time step for the evolution), and are computationally
+    faster.
+
+    The algorithm and its theoretical derivation are described in [1]_.
+
+    References
+    ----------
+    .. [1] A Morphological Approach to Curvature-based Evolution of Curves and
+           Surfaces, Pablo Márquez-Neila, Luis Baumela, Luis Álvarez. In IEEE
+           Transactions on Pattern Analysis and Machine Intelligence (PAMI),
+           2014, DOI 10.1109/TPAMI.2013.106
+    """
+
+    init_level_set = _init_level_set(init_level_set, image.shape)
+
+    _check_input(image, init_level_set)
+
+    u = np.int8(init_level_set > 0)
+
+    iter_callback(u)
+
+    for _ in range(iterations):
+
+        # inside = u > 0
+        # outside = u <= 0
+        c0 = (image * (1 - u)).sum() / float((1 - u).sum() + 1e-8)
+        c1 = (image * u).sum() / float(u.sum() + 1e-8)
+
+        # Image attachment
+        du = np.gradient(u)
+        abs_du = np.abs(du).sum(0)
+        aux = abs_du * (lambda1 * (image - c1)**2 - lambda2 * (image - c0)**2)
+
+        u[aux < 0] = 1
+        u[aux > 0] = 0
+
+        # Smoothing
+        for _ in range(smoothing):
+            u = _curvop(u)
+
+        iter_callback(u)
+
+    return u
+
+
+def morphological_geodesic_active_contour(gimage, iterations,
+                                          init_level_set='circle', smoothing=1,
+                                          threshold='auto', balloon=0,
+                                          iter_callback=lambda x: None):
+    """Morphological Geodesic Active Contours (MorphGAC).
+
+    Geodesic active contours implemented with morphological operators. It can
+    be used to segment objects with visible but noisy, cluttered, broken
+    borders.
+
+    Parameters
+    ----------
+    gimage : (M, N) or (L, M, N) array
```

Comment:
Our naming convention says this should be `image`

Reply:
This is on purpose. This argument is rarely the image the user wants to work with, but a transformation of that image as explained in the docstring of the function. I called it `gimage` instead of `image` to hint the user about reading the docstring. It can be changed if the naming convention is critical, though.